### PR TITLE
feat: added go-client wrapper for error reporting

### PR DESCRIPTION
**Pull request by madhuravius**
This should properly result in better error messages.

- [x] Unit tests
- [x] Proof and use in common flows (screenshots below, remain the same in more recent testing)
- [x] Testing common flows end to end and validating across the board (tested all major endpoints/resources and still doing so)

---

Before: 

image

Wrapped errors with changes:

image

image

image

### CODE REVIEW

File: aptible/resource_endpoint.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func resourceEndpointCreate(d *schema.ResourceData, meta interface{}) error {
 	endpointType, err := aptible.GetEndpointType(humanReadableEndpointType)
 	if err != nil {
 		log.Println(err)
-		return err
+		return generateErrorFromClientError(err)
```
Comment (madhuravius) [inline aptible/resource_endpoint.go:146]: incorrect usage, revert

File: aptible/client_utils.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	if err != nil {
+		return fmt.Errorf("Unable to properly decode error in marshal from client - %s\n", err.Error())
+	}
+
+	var out clientError
+	if err := json.Unmarshal(data, &out); err != nil {
+		return fmt.Errorf("Unable to properly decode error in unmarshal from client - %s\n", err.Error())
+	}
+
+	if out.Payload.Code == nil || out.Payload.Error == nil {
```
Comment (joshraker) [inline aptible/client_utils.go:44]: Why is `out.Payload.Message` not included in this check?

**Review by joshraker — APPROVED** [review]
All of the changes look good. I did not pull down the repo to search for places that should be using the new client method that may have been missed.

**Comment by madhuravius** [comment]
> All of the changes look good. I did not pull down the repo to search for places that should be using the new client method that may have been missed.

I checked again on Tuesday, but I made sure to wrap every go-deploy client method (except 2 - GetApp and SetupClient) in the terraform provider with that code. I left those two out and planned on adding them in at some point later if needed/desired (both had additional wraps/guards within client calls compared to other methods and they did not cleanly bubble up the underlying client errors).

**Review by almathew — APPROVED** [review]

**Pull request merged**